tf2_py/src/tf2_py.cpp

```diff
@@ -265,17 +279,19 @@ static PyObject *getLatestCommonTime(PyObject *self, PyObject *args)
   tf2::BufferCore *bc = ((buffer_core_t*)self)->bc;
   char *target_frame, *source_frame;
   tf2::CompactFrameID target_id, source_id;
-  builtin_interfaces::msg::Time time;
+  tf2::TimePoint time;
   std::string error_string;
 
   if (!PyArg_ParseTuple(args, "ss", &target_frame, &source_frame))
     return NULL;
   WRAP(target_id = bc->_validateFrameId("get_latest_common_time", target_frame));
   WRAP(source_id = bc->_validateFrameId("get_latest_common_time", source_frame));
-  int r = bc->_getLatestCommonTime(target_id, source_id, time, &error_string);
-  if (r == 0) {
+  auto r = bc->_getLatestCommonTime(target_id, source_id, time, &error_string);
```

Comment:
This change to "auto" (and elsewhere in the file) again look like something that could go into upstream tf2_py.